### ISSUE TITLE
Fedora Developer Portal package

### ISSUE THREAD

**Issue by frantisekz** [developer-portal/website#41]
It'll be great to have Fedora Developer Portal packaged as application in Fedora.
There is long way to go and I'll try to track progress in this issue.

If you interested, here is really early version: https://github.com/frantisekz/content/releases/download/temp/fedora-developer-portal-0.0.1-1.x86_64.rpm
(Can be launched with `fedora-developer-portal` command)

Before we can try to get this reviewed and included in Fedora repositories we need to have Electron packaged in Fedora. I'll update this package from time to time and I'll try to setup copr repository with this during next week.

**Comment by frantisekz** [comment]
I was able to fix paths issues relatively easily and I've created repo with fixed version: https://github.com/frantisekz/fedora-developer-portal-portable

So, the next things on todo are:
*Update rpm to use portable version on Fedora developer portal (so it could be used offline)
*Unbundle Electron from app itself so it could be imported to COPR
*Find somebody to package Electron to Fedora

**Comment by phracek** [comment]
What about to create the package Fedora Developer Portal also in COPR? At least for testing.

**Comment by frantisekz** [comment]
@phracek Yep, I am working on it. I have only basic knowledge about packaging, so it is taking a bit longer.

**Comment by frantisekz** [comment]
@phracek Here you go https://copr.fedorainfracloud.org/coprs/frantisekz/fedora-developer-portal/

(No .desktop file yet, use "fedora-developer-portal" in terminal)

**Comment by phracek** [comment]
I have tried it and it really works. Application is opened and I see Fedora Developer Portal as a application. Awesome.

**Comment by frantisekz** [comment]
I've released new version of fedora developer portal app for Fedora 24 and Rawhide. It doesn't depend on Electron, but on Epiphany. Thanks @mcatanzaro !

**Comment by phracek** [comment]
@frantisekz And what about F23?

**Comment by frantisekz** [comment]
@phracek F23 won't be supported since it depends on at least Epiphany 3.20, sorry.

**Comment by frantisekz** [comment]
Review request: https://bugzilla.redhat.com/show_bug.cgi?id=1341662

**Comment by mcatanzaro** [comment]
I would hold off a bit before pushing this to Fedora, because we discovered this morning that it probably only works in GNOME and associated desktops as it depends on the GIO_LAUNCHED_DESKTOP_FILE environment variable... and fixing this might require incompatible changes. See [GNOME #767101](https://bugzilla.gnome.org/show_bug.cgi?id=767101).

**Comment by frantisekz** [comment]
@mcatanzaro I'll test it in some different DE and it might be possible to workaround potential issues by calling fedora-developer-portal script from .desktop file which sets GIO_LAUNCHED_DESKTOP_FILE .

**Comment by mcatanzaro** [comment]
Yes, a workaround using a script should work fine... but it really shouldn't be required. To my knowledge, you're the first person to try distributing a system-wide GNOME web app, so we can afford to make an incompatible change to get this right.

**Comment by frantisekz** [comment]
I personally have only GNOME but I tested it quickly on different desktop and even script doesn't work. I'll try to play with it.

As I am thinking about this, it might be better to break everything now instead of having different behaviour between 3.20 and 3.22.

**Comment by pvalena** [comment]
It'd be nice to have the website built in Koji and then the included tests ran. Let me know the state of this.

**Comment by frantisekz** [comment]
@pvalena I've enabled tests in koji [0]. There are 6 Failures [1].

[0] https://src.fedoraproject.org/rpms/fedora-developer-portal/c/8d97e944351d9136932447eb868d7d6e886b21a4?branch=master
[1] https://kojipkgs.fedoraproject.org//work/tasks/7429/27617429/build.log

**Comment by pvalena** [comment]
Thanks!

There's a work in progress to fix those tests. Could you do the releases on the basis of the [actual releases](https://github.com/developer-portal/developer.fedoraproject.org/commits/release)?

**Comment by pvalena** [comment]
Note: you probably do not need the website running (`--detach`) to be able to run the tests.
